Fixtures/Miscellaneous/Plugins/ContrivedTestPlugin/Plugins/MyAmbiguouslyNamedCommandPlugin/plugin.swift

```diff
@@ -1,26 +1,33 @@
 import PackagePlugin
  
-for inputFile in targetBuildContext.inputFiles.filter({ $0.path.extension == "dat" }) {
-    let inputPath = inputFile.path
-    let outputName = "Ambiguous_" + inputPath.stem + ".swift"
-    let outputPath = targetBuildContext.pluginWorkDirectory.appending(outputName)
-    commandConstructor.addBuildCommand(
-        displayName:
-            "This is a constant name",
-        executable:
-            try targetBuildContext.tool(named: "MySourceGenBuildTool").path,
-        arguments: [
-            "\(inputPath)",
-            "\(outputPath)"
-        ],
-        environment: [
-            "VARIABLE_NAME_PREFIX": "SECOND_PREFIX_"
-        ],
-        inputFiles: [
-            inputPath,
-        ],
-        outputFiles: [
-            outputPath
-        ]
-    )
+@main struct MyPlugin: BuildToolPlugin {
```

Comment:
what is the API convention for @main: on top or at the same level? ie

`@main struct MyPlugin: BuildToolPlugin`

or

```
@main 
struct MyPlugin: BuildToolPlugin {
```

Reply:
Don't know, but the latter looked silly to me.  But if it's convention we'll of course use it.

Reply:
Updated